backend/baseclass.pm

```diff
@@ -838,9 +839,11 @@ previous test's serial output. Call this before you start doing something new
 =cut
 
 sub set_serial_offset {
-    my ($self, $args) = @_;
+    my ($self, $keep) = @_;
```

Comment:
With the method's name, I'd expect that the (only) argument is the offset, not some weird "keep" value which is substracted for an (undocumented) reason

Reply:
Also, if you keep $self and you are calling this function `$self->set_serial_offset($bytes_to_keep);`... not sure how it will work, cause it stores the value of `$bytes_to_keep` in $self... 
So, either you really use %args and call the function like  `$self->set_serial_offset(keep => $bytes_to_keep);`
or you just  do `my ($keep) = @_;`